### PR TITLE
Update to 3.2.1 and build PPC64LE ARM64

### PR DESCRIPTION
**Pull request by wolfv**
…nda-forge-pinning 2020.10.04.20.30.50

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
Based on the CI, I guess the macOS build is blocked by https://github.com/conda-forge/libglu-feedstock/pull/17 .

**Comment by traversaro** [comment]
> or perhaps we libglu is not actually a dependency on macOS, I am not an expert of freeglut

Indeed, in the homebrew formula of freeglut ( https://github.com/Homebrew/homebrew-core/blob/masster/Formula/freeglut.rb ) and the vcpkg port ( https://github.com/microsoft/vcpkg/tree/master/ports/freeglut ) seems to only depend on X11, but to be honest I am not sure of the relation between X11 and the libglu library.

**Comment by traversaro** [comment]
> 
> 
> > or perhaps we libglu is not actually a dependency on macOS, I am not an expert of freeglut
> 
> Indeed, in the homebrew formula of freeglut ( https://github.com/Homebrew/homebrew-core/blob/masster/Formula/freeglut.rb ) and the vcpkg port ( https://github.com/microsoft/vcpkg/tree/master/ports/freeglut ) seems to only depend on X11, but to be honest I am not sure of the relation between X11 and the libglu library.

Looking more in depth both homebrew and vcpkg has a patch to let freeglut use XQuartz on macOS, probably it could make sense to import it also here: 
* https://github.com/Homebrew/homebrew-core/blob/9591758fd8a565abf98d13ffd87c66e5e12ad579/Formula/freeglut.rb#L33
* https://github.com/microsoft/vcpkg/blob/2020.07/ports/freeglut/macOS_Xquartz.patch

**Comment by traversaro** [comment]
Indeed, I looked a bit more into this and apparently macOS ships its own system version of GLUT libraries as part of its OpenGL support, that even if deprecated will continue to be supported on Big Sur/Apple Silicon (see https://developer.apple.com/forums/thread/650427). So, even if one wants to build freeglut, I guess that glu and all the related OpenGL headers can be provided by the system OpenGL.

**Comment by hmaarrfk** [comment]
Seems like the name of the windows header changed.

Is it safe to skip osx again?

**Comment by Tobias-Fischer** [comment]
I'd say it's safe to skip osx; as @traversaro mentioned the system version of GLUT is probably preferable on OSX.

**Comment by wolfv** [comment]
I fixed windows, and skipped os x again

**Comment by Tobias-Fischer** [comment]
The windows test still fails - this should be an easy fix though, maybe just the name of the library needs changing?

**Comment by wolfv** [comment]
hnng so you can choose:

glut-compatible: glut.h, libname = glut.lib / dll
freeglut: No glut.h, libname = freeglut.lib / dll

Before it was glut.h + freeglut.lib ... 

@jakirkham any input?

**Comment by Tobias-Fischer** [comment]
Hi @jakirkham @wolfv, do you have any further idea about the lib/include names? It would be great to see this PR merged so we can further advance ARM64 :)

**Comment by wolfv** [comment]
I pinged people on gitter, let's see if we get any advice.

**Comment by isuruf** [comment]
Changing the dll name will break downstream packages.

**Comment by jakirkham** [comment]
Sorry Wolf I don't have deep thoughts on Windows. Would defer to others :)

**Comment by wolfv** [comment]
@isuruf how about adding a symlink?

**Comment by Tobias-Fischer** [comment]
Quick bump @wolfv @isuruf

**Comment by wolfv** [comment]
I tested that jasper and fsleyes still work after these changes here and they do. I think this is ready to go @jakirkham

**Comment by wolfv** [comment]
Feel free to squash. Thx

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,45 +7,50 @@ package:
...
   fn: freeglut-{{ version }}.tar.bz2
   url: http://sourceforge.net/projects/freeglut/files/freeglut/{{ version }}/freeglut-{{ version }}.tar.gz
-  md5: 90c3ca4dd9d51cf32276bc5344ec9754
+  sha256: d4000e02102acaf259998c870e25214739d1f16f67f99cb35e4f46841399da68
 
 build:
-  number: 1005
-  skip: true                      # [osx]
+  number: 0
+  skip: true  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml:14]: It looks like this is still skipped. Is that expected?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,45 +7,50 @@ package:
...
-    - test -d "${PREFIX}/include/GL"  # [linux]
+    - test -d "${PREFIX}/include/GL"  # [unix]
 
       # Test libraries.
-    - test -f "${PREFIX}/lib/libglut.a"   # [linux]
-    - test -f "${PREFIX}/lib/libglut.so"  # [linux]
-    - ldd "${PREFIX}/lib/libglut.so"      # [linux]
+    - test ! -f "${PREFIX}/lib/libglut.a"   # [unix]
+    - test -f "${PREFIX}/lib/libglut.so"    # [unix]
+    - ldd "${PREFIX}/lib/libglut.so"        # [unix]
```
Comment (jakirkham) [inline recipe/meta.yaml:53]: macOS doesn't use `.so`s. It produces `.dylib`s. So I'm not sure this makes sense to test outside of Linux.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -7,45 +7,50 @@ package:
...
-    - xorg-libxau                        # [linux]
-    - xorg-libxext                       # [linux]
-    - xorg-libxfixes                     # [linux]
-    - xorg-libxi                         # [linux]
+    - libxcb                             # [unix]
+    - xorg-libx11                        # [unix]
+    - xorg-libxau                        # [unix]
+    - xorg-libxext                       # [unix]
+    - xorg-libxfixes                     # [unix]
+    - xorg-libxi                         # [unix]
```
Comment (jakirkham) [inline recipe/meta.yaml:43]: It's unclear to me whether we would need these on macOS as it has it's own GL support as part of the OS.

**Review by jakirkham — COMMENTED** [review]
Thanks Wolf!

I'm a little confused about the macOS changes. Have commented below. Also it appears we are not actually building on macOS (even though that's in the PR title). Not sure if that's leftover or if I'm missing something

File: recipe/meta.yaml (hunk 1 above)
Comment (wolfv) [reply recipe/meta.yaml:14]: Yeah, expected :)

**Comment by wolfv** [comment]
you're right, we determined that it's not worth trying to build this on OS X as the system ships with GLUT. 

I don't think anything in this recipe is causing any harm though, so I still consider it good to merge?

File: recipe/meta.yaml (hunk 2 above)
Comment (wolfv) [reply recipe/meta.yaml:53]: you're right, but we never actually got OS X to build.

This still contains an update for FreeGLUT as well as builds for ARM64 and PPC64LE

Comment (jakirkham) [reply recipe/meta.yaml:53]: Those would already be covered under `linux` though. Any reason not to just stick with that?

**Comment by jakirkham** [comment]
I'd prefer not to include any macOS related changes given we are not building it. Could you please revert them?

**Comment by wolfv** [comment]
@jakirkham done.

**Review by jakirkham — APPROVED** [review]

**Pull request merged**